addon/-private/system/class-meta.js

```diff
@@ -0,0 +1,57 @@
+import EmptyObject from 'ember-data/-private/system/empty-object';
+
+export default class ClassMeta {
```

Comment:
Outside of one new behavior, this object mirrors the pojo it replaces. Don't disagree about unit tests, I just suspect we already implicitly have a bunch that we should congregate and expand upon.